client/components/AdvancedSearch/index.tsx

```diff
@@ -102,6 +142,29 @@ export class AdvancedSearch extends React.PureComponent<IProps> {
                 schema: {},
             },
             {
+                ...customTextFields,
+                user: {
+                    field: 'userIds',
+                    valueStoredAsArray: true,
+                    deskId: this.props.filterUsersByDesk,
+                },
+                content_type: {
+                    field: 'contentType',
+                },
+                multiple_content: {
+                    field: 'multipleContent',
+                },
+                genre: {
+                    clearable: true,
+                },
+                assignment_priority: {
+                    field: 'assignmentPriority',
+                    noTitleInPopup: true,
+                    clearable: true,
+                },
+                anpa_category: {
+                    field: this.props.type === 'assignments' ? 'anpaCategory' : 'anpa_category',
```

Comment:
In many places you use a ternary where it should be a switch instead - even better if in combination with `assertNever`. Or multiple `if` statements [like here](https://github.com/superdesk/superdesk-planning/blob/feature/embedded-events-editor-async/client/components/editor-standalone/authoring-storage-in-memory.ts#L14).

At the moment it's not causing issues since there are only 2 possible string values, but adding more is much easier when using `assertNever` since one only needs to follow compiler errors to find all places that need to be updated.